### PR TITLE
Review  code

### PR DESCRIPTION
**Pull request by BaraahAlKalbani**
Currently the Tic Tac Toe doesn’t retain the state if the uses closes the game while the gameplay is in progress. User should have the ability to resume the game from the last saved state.

Acceptance Criteria:

Allow the player to exit the game before completing it.
Next time the application runs, it should see if there is a saved in-progress game and should prompt the user to continue where they left OR start new game.
If the user selects continue, then it should resume the game.
‌

### CODE REVIEW

**Review by iamabdh — APPROVED** [review]

File: src/SaveAndLoad.java [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+            }
+
+            BufferedWriter writer = new BufferedWriter(new FileWriter(fileName));
+            // Write the game state to the file
+            writer.write(String.valueOf(player1) + "\n");
+            writer.write(String.valueOf(player2) + "\n");
+            writer.write(currentPlayer + "\n");
+            writer.write(turn + "\n");
+            writer.write("Board:\n");
+            for (int i = 0; i < board.length; i++) {
```
Comment (FatemaAlbalushi) [inline src/SaveAndLoad.java:54]: good job but instead of having i and j try to use meaningful name for example row and col

**Review by FatemaAlbalushi — COMMENTED** [review]
See the comment:

**Review by FatemaAlbalushi — COMMENTED** [review]
Great work

**Review by FatemaAlbalushi — APPROVED** [review]
Great job

File: src/PlayTurn.java [hunk 2]
```diff
@@ -1,75 +1,94 @@
...
+        System.out.println("Player " + playerNo + " YOUR SYMBOL IS:" + playerSymble + "\nIs that Correct?");
+        char confirm = scan.next().charAt(0);
+        // Confirming player Symbol
+        if (confirm == 'n' || confirm == 'N') {
+            System.out.println("Player " + playerNo + " ENTER YOUR NEW SYMBOL:");
             playerSymble = scan.next().charAt(0);
         }
     }
-    
-
```
Comment (AnwaarAlbusaidi) [inline src/PlayTurn.java:22]: remove spaces

File: src/PlayTurn.java [hunk 3]
```diff
@@ -1,75 +1,94 @@
...
+     * @param playernumber :the number of the playing player
+     */
+    public static void playerTurn(char[][] block, char playerSymble, boolean gameWon, int turn, int playernumber) {
+
+        System.out.println("Player " + playernumber + " Enter Your Move:");
         Scanner scan = new Scanner(System.in);
         char input = scan.next().charAt(0);
         boolean inputFound = false;
-        
-        //do while loop will repeat until the player input an available/valid position
```
Comment (AnwaarAlbusaidi) [inline src/PlayTurn.java:31]: repeat the same comment, remove one

**Review by AnwaarAlbusaidi — APPROVED** [review]
good job, all the best.

File: src/PlayTurn.java (hunk 2 above)
Comment (BaraahAlKalbani) [reply src/PlayTurn.java:22]: red mean the old file not the pull requested file there is no spaces in the code in this lines

File: src/PlayTurn.java (hunk 3 above)
Comment (BaraahAlKalbani) [reply src/PlayTurn.java:31]: red mean the old file not the pull requested file there is no repeated lines in the code

**Pull request merged**